test/test-network/systemd-networkd-tests.py

```diff
@@ -0,0 +1,857 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: LGPL-2.1+
+# systemd-networkd tests
+
+import os
+import sys
+import unittest
+import subprocess
+import time
+import shutil
+import signal
+import socket
+import threading
+from shutil import copytree
+
+network_unit_file_path='/var/run/systemd/network'
+networkd_ci_path='/var/run/networkd-ci'
+network_sysctl_ipv6_path='/proc/sys/net/ipv6/conf'
+network_sysctl_ipv4_path='/proc/sys/net/ipv4/conf'
+
+dnsmasq_config_file='/var/run/networkd-ci/test-dnsmasq.conf'
+dnsmasq_pid_file='/var/run/networkd-ci/test-test-dnsmasq.pid'
+dnsmasq_log_file='/var/run/networkd-ci/test-dnsmasq-log-file'
+
+def setUpModule():
+
+    os.makedirs(network_unit_file_path, exist_ok=True)
+
+    shutil.rmtree(networkd_ci_path)
+    copytree('conf', networkd_ci_path)
+
+def tearDownModule():
+    shutil.rmtree(networkd_ci_path)
+
+class Utilities():
+    dhcp_server_data = []
+
+    def read_link_attr(self, link, dev, attribute):
```

Comment:
Why does this (and all following methods in this class) need to be a class method?

Reply:
Because none of them does anything with the class object (the `self` argument).

Reply:
The intent is to keep methods together.